src/lib/storage/base_dictionary_segment.hpp

```diff
@@ -50,5 +50,7 @@ class BaseDictionarySegment : public AbstractEncodedSegment {
    * @brief Returns encoding specific null value ID
    */
   virtual ValueID null_value_id() const = 0;
+
+  virtual bool is_nullable() const = 0;
```

Comment:
Why is this part of the Base**Dictionary**Segment? Shouldn't all encode be able to fill this flag in any type of encoded segment?

Reply:
We would need to touch all encoders (doable), but the main reason was that I don't think we would specialize other iterables. It would probably hurt compilation time and no one -- except me -- regularly uses other encodings.

Reply:
We can still hold short of touching the other iterables.

Reply:
You mean we put the flag/getters into the base segments but don't touch the other encoders and iterables?

Reply:
Yes.